crates/evm/evm/src/executors/invariant/mod.rs

```diff
@@ -241,6 +241,15 @@ impl InvariantTest {
         }
         self.test_data.fuzz_cases.push(FuzzedCases::new(run.fuzz_runs));
 
+        // Prune older cases if we have too many.
+        // We keep a rolling window of full traces for the last 4096 runs to aid debugging,
+        // but prune older ones to avoid unbounded memory usage.
+        const MAX_KEPT_CALLDATA: usize = 4096;
+        if self.test_data.fuzz_cases.len() > MAX_KEPT_CALLDATA {
+            let prune_index = self.test_data.fuzz_cases.len() - MAX_KEPT_CALLDATA - 1;
+            self.test_data.fuzz_cases[prune_index].prune_calldata();
+        }
+
```

Comment:
This doesn't actually shrink the `fuzz_cases` vec. Maybe we should use a ring buffer or altogether disable this for long running campaigns

cc @grandizzy

Reply:
Or vec deque

```
use std::collections::VecDeque;

if self.test_data.fuzz_cases.len() >= 4096 {
    self.test_data.fuzz_cases.pop_front();
}
self.test_data.fuzz_cases.push_back(new_case);
```